roles/ceph-osd/tasks/crush_rules.yml

```diff
@@ -24,7 +24,7 @@
   with_items: "{{ hostvars[groups[mon_group_name][0]]['crush_rules'] | unique }}"
   delegate_to: '{{ groups[mon_group_name][0] }}'
   run_once: true
-  when: item.default | bool
+  when: item.default is defined | bool
```

Comment:
we should always have the `default` key defined in the `crush_rules` dict

Reply:
I agree. I made the change to use the default(omit) filter, which will make it so and not fail if you do what I did and leave the key commented out by mistake.
 
Also, do you know where this is documented? upstream/or downstream. 

The only reason I configured crush_rules was because of the [condition](https://github.com/ceph/ceph-ansible/blob/master/roles/ceph-osd/tasks/main.yml#L90). All I really wanted to do was configure crush hierarchy.

Issue with defaults is the expectation of both [classes](https://github.com/ceph/ceph-ansible/blob/master/roles/ceph-mon/defaults/main.yml#L47) ... which leads to issues [here](https://github.com/ceph/ceph-ansible/blob/master/roles/ceph-osd/tasks/crush_rules.yml#L16)